src/BloomExe/web/EnhancedImageServer.cs

```diff
@@ -140,8 +141,75 @@ public static SimulatedPageFile MakeSimulatedPageFileInBookFolder(HtmlDom dom, b
 			return new SimulatedPageFile() {Key = url};
 		}
 
+		/// <summary>
+		/// Producing PDF with gecko doesn't work on Windows for some reason when referencing
+		/// the localhost server.  So we need to use actual files, and refer to actual files in
+		/// the links to the css files.
+		/// </summary>
+		/// <remarks>
+		/// See http://jira.sil.org/browse/BL-932 for details on the bug.
+		/// </remarks>
+		public static SimulatedPageFile MakeRealPublishModeFileInBookFolder(HtmlDom dom)
+		{
+			var simulatedPageFileName = Path.ChangeExtension(Guid.NewGuid().ToString(), ".htm");
+			var pathToSimulatedPageFile = simulatedPageFileName; // a default, if there is no special folder
+			if (dom.BaseForRelativePaths != null)
+				pathToSimulatedPageFile = Path.Combine(dom.BaseForRelativePaths, simulatedPageFileName).Replace('\\', '/');
+			pathToSimulatedPageFile = RemoveLocalhostReferenceFromPath(pathToSimulatedPageFile);
+			FixStyleLinkReferences(dom);
+			var html5String = TempFileUtils.CreateHtml5StringFromXml(dom.RawDom);
+			using (var writer = File.CreateText(pathToSimulatedPageFile))
+			{
+				writer.Write(html5String);
+				writer.Close();
+			}
+			var url = pathToSimulatedPageFile.Replace('\\', '/');
+			if (Regex.IsMatch(pathToSimulatedPageFile, "^[A-Z]:/"))
+				url = "file:///" + url;
+			else
+				url = "file://" + url;
+			return new SimulatedPageFile() { Key = url };
+		}
+
+		private static string RemoveLocalhostReferenceFromPath(string path)
```

Comment:
There's already a utility method somewhere like FromLocalHost...does this do something different?